File: algorithms/math/euler-toient/swift/EulerTotient.swift
func eulerTotient(_ n: Int) -> Int {
    if n <= 0 { return 0 }
    if n == 1 { return 1 }

    var result = n
    var value = n
    var factor = 2

    while factor * factor <= value {
        if value % factor == 0 {
            while value % factor == 0 {
                value /= factor
            }
            result -= result / factor
        }
        factor += 1
    }

    if value > 1 {
        result -= result / value
    }

    return result
}

File: algorithms/math/euler-toient/rust/euler_totient.rs
pub fn euler_totient(n: i64) -> i64 {
    if n <= 1 {
        return 1;
    }

    let mut value = n;
    let mut result = n;
    let mut factor = 2i64;

    while factor * factor <= value {
        if value % factor == 0 {
            while value % factor == 0 {
                value /= factor;
            }
            result -= result / factor;
        }
        factor += 1;
    }

    if value > 1 {
        result -= result / value;
    }

    result
}
